lib/Controller/DocumentController.php

```diff
@@ -116,6 +116,13 @@ public function index($fileId) {
 				'instanceId' => $this->settings->getSystemValue('instanceid'),
 			];
 
+			// Update the current file to be accessible with system public shared key
+			$owner = $item->getOwner()->getUID();
+			$absPath = '/' . $owner . '/' .  $item->getInternalPath();
+			$accessList = \OC::$server->getEncryptionFilesHelper()->getAccessList($absPath);
+			$accessList['public'] = true;
+			\OC::$server->getEncryptionManager()->getEncryptionModule()->update($absPath, $owner, $accessList);
```

Comment:
While the document is being edited, the file has already been downloaded by Collabora Online. Any further attempts to open the same file in the nextcloud by any user will not ensue a download request again to the nextcloud server as long as the document is opened in Collabora Online; other users just join that same document. So, even if the access list is recalculated for any reason depriving public access to the file, editing session should keep working.

The window between when this patch of code is executed and when Collabora Online makes a request to download the file is ephemeral. If public access is stripped off in this short window, yeah, we would be unable to get the encrypted file but chances of this happening are less.

Having said that, I am convinced that implementing a small private share provider as part of documents app might be a better approach but for the time being, I also don't see any major repercussions with the current approach.

Do you know if there's already some app that implements a private share provider ?